app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,338 @@
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * List item used to display a label and an icon at the beginning with an optional description
+ * text and an optional, interactable icon at the end.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param drawable [DrawableRes] used to display a drawable at the start of the list item.
+ * @param drawableDescription Content description of the drawable.
+ * @param icon Optional [DrawableRes] used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Called when the user clicks on the icon.
+ */
+@Composable
+fun IconListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    @DrawableRes drawable: Int,
+    drawableDescription: String? = null,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        beforeListAction = {
+            ListItemIcon(
+                icon = drawable,
+                iconDescription = drawableDescription,
+                onClick = null,
+            )
+        },
+        afterListAction = {
+            icon?.let {
+                ListItemIcon(
+                    icon = icon,
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+            }
+        }
+    )
+}
+
+/**
+ * Base list item used to display a label with an optional description text and
+ * the flexibility to add custom UI to either end of the item.
+ *
+ * @param label The label in the list item.
+ * @param description An optional description text below the label.
+ * @param onClick Called when the user clicks on the item.
+ * @param beforeListAction Optional Composable for adding UI to the start of the list item.
+ * @param afterListAction Optional Composable for adding UI to the end of the list item.
+ */
+@Composable
+private fun ListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    beforeListAction: @Composable RowScope.() -> Unit = {},
+    afterListAction: @Composable RowScope.() -> Unit = {},
+) {
+    Row(
+        modifier = when (onClick != null) {
+            true -> Modifier.clickable { onClick() }
+            false -> Modifier
+        }.then(
+            Modifier.defaultMinSize(minHeight = LIST_ITEM_HEIGHT)
+        ),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        beforeListAction()
+
+        Column(
+            modifier = Modifier
+                .padding(horizontal = 16.dp, vertical = 10.dp)
```

Comment:
`10.dp` seems a bit arbitrary for such a scenario. I would even prefer to just specify `6.dp` since that was the minimum padding for the label + description scenario.

Reply:
That sounds good to me!